safe/impact_function/multi_exposure_wrapper.py

```diff
@@ -0,0 +1,225 @@
+    ANALYSIS_FAILED_BAD_INPUT,
+    ANALYSIS_SUCCESS,
+)
+from safe.impact_function.impact_function import ImpactFunction
+from safe.impact_function.impact_function_utilities import check_input_layer
+from safe.gui.widgets.message import generate_input_error_message
+from safe.utilities.gis import deep_duplicate_layer
+from safe.utilities.i18n import tr
+from safe import messaging as m
+
+LOGGER = logging.getLogger('InaSAFE')
+
+
+class MultiExposureImpactFunction(object):
+
+    """Multi-exposure wrapper."""
+
+    def __init__(self):
+        """Constructor."""
+        # Input layers
+        self._hazard = None
+        self._aggregation = None
+
+        # Exposures, it's now a list of layers. One for each exposure maximum.
+        self._exposures = []
+
+        # For now, we have many IF running.
+        self._impact_functions = []
+        self._if_being_processed = None
+
+        # Metadata
+        self.callback = None
+        self.debug = False
+        self._is_ready = False
+        self._start_datetime = None
+        self._end_datetime = None
+        self._duration = 0
+
+    @property
+    def hazard(self):
+        """Property for the hazard layer to be used for the analysis.
+
+        :returns: A map layer.
+        :rtype: QgsMapLayer
+        """
+        return self._hazard
+
+    @hazard.setter
+    def hazard(self, layer):
+        """Setter for hazard layer property.
+
+        :param layer: Hazard layer to be used for the analysis.
+        :type layer: QgsMapLayer
+        """
+        self._hazard = layer
+        self._is_ready = False
+
+    @property
+    def exposures(self):
+        """Property for exposure layers to be used for the analysis.
+
+        :returns: List of map layers.
+        :rtype: list(QgsMapLayer)
+        """
+        return self._exposures
+
+    @exposures.setter
+    def exposures(self, layers):
+        """Setter for exposure layers property.
+
+        :param layers: List of exposure layers to be used for the analysis.
+        :type layers: list(QgsMapLayer)
+        """
+        self._exposures = layers
+        self._is_ready = False
+
+    @property
+    def aggregation(self):
+        """Property for the aggregation layer to be used for the analysis.
+
+        :returns: A map layer.
+        :rtype: QgsVectorLayer
+        """
+        return self._aggregation
+
+    @aggregation.setter
+    def aggregation(self, layer):
+        """Setter for aggregation layer property.
+
+        :param layer: aggregation layer to be used for the analysis.
+        :type layer: QgsVectorLayer
+        """
+        self._aggregation = layer
+        self._is_ready = False
+
+    @property
+    def impact_functions(self):
+        """Return the list of impact functions which have been used.
+
+        :return: List of impact functions.
+        :rtype: list(ImpactFunction)
+        """
+        return self._impact_functions
+
+    @property
+    def if_being_processed(self):
```

Comment:
The first time I read, I thought it's like a checker/validation method. Perhaps change it to more easy to read (e.g. `current_impact_function`, or another name :D )

Reply:
Good catch, you're right.